lighthouse-core/audits/preload-lcp.js

```diff
@@ -0,0 +1,185 @@
+  description: 'Consider preloading the image used by ' +
+    'the LCP element in order to improve your LCP time.',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class PreloadLCPAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'preload-lcp',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs', 'URL', 'TraceElements'],
+      scoreDisplayMode: Audit.SCORING_MODES.NUMERIC,
+    };
+  }
+
+  /**
+   *
+   * @param {LH.Artifacts.NetworkRequest} request
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {Array<LH.Gatherer.Simulation.GraphNode>} initiatorPath
+   * @param {string|undefined} lcpImageSource
+   * @return {boolean}
+   */
+  static shouldPreloadRequest(request, mainResource, initiatorPath, lcpImageSource) {
+    const mainResourceDepth = mainResource.redirects ? mainResource.redirects.length : 0;
+
+    // If it's not the request for the LCP image, don't recommend it.
+    if (request.url !== lcpImageSource) return false;
+    // If it's already preloaded, no need to recommend it.
+    if (request.isLinkPreload) return false;
+    // It's not a request loaded over the network, don't recommend it.
+    if (URL.NON_NETWORK_PROTOCOLS.includes(request.protocol)) return false;
+    // It's not at the right depth, don't recommend it.
+    if (initiatorPath.length < mainResourceDepth + 1) return false;
+    // Finally, return whether or not it belongs to the main frame
+    return request.frameId === mainResource.frameId;
+  }
+
+  /**
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @param {LH.Artifacts.TraceElement|undefined} lcpElement
+   * @return {string|null}
+   */
+  static getURLToPreload(mainResource, graph, lcpElement) {
+    if (!lcpElement) {
+      return null;
+    }
+
+    let lcpUrl = null;
+    graph.traverse((node, traversalPath) => {
+      if (node.type !== 'network') return;
+      const record = node.record;
+      const imageSource = lcpElement.imageSource;
+      // Don't include the node itself or any CPU nodes in the initiatorPath
+      const path = traversalPath.slice(1).filter(initiator => initiator.type === 'network');
+      if (!PreloadLCPAudit.shouldPreloadRequest(record, mainResource, path, imageSource)) return;
+      lcpUrl = node.record.url;
+    });
+
+    return lcpUrl;
+  }
+
+  /**
+   * Computes the estimated effect of preloading the LCP image.
+   * @param {string} lcpUrl // The image URL of the LCP
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @param {LH.Gatherer.Simulation.Simulator} simulator
+   * @return {{wastedMs: number, results: Array<{url: string, wastedMs: number}>}}
+   */
+  static computeWasteWithGraph(lcpUrl, graph, simulator) {
```

Comment:
seems like this function isn't really necessary for this phase of the PR

or just replace all this with `return Array.from(simulation.nodeTimings.entries()).find(([node, timing]) => node.record.url === lcpUrl)[1].duration`